guides/source/action_view_helpers.md

```diff
@@ -5,668 +5,1054 @@ Action View Helpers
+# => <link href="/assets/application.css" media="all" rel="stylesheet" />
 ```
 
-See the [API Documentation](https://api.rubyonrails.org/classes/ActionView/Helpers/JavaScriptHelper.html#method-i-javascript_tag) for more information.
+`media` is used to specify the media type for the link. The most common media
+types are `all`, `screen`, `print`, and `speech`.
 
-### NumberHelper
+INFO: Internally, `stylesheet_link_tag` uses [`stylesheet_path` from the
+AssetUrlHelpers](https://api.rubyonrails.org/classes/ActionView/Helpers/AssetUrlHelper.html#method-i-stylesheet_path)
+to build the stylesheet path.
 
-Provides methods for converting numbers into formatted strings. Methods are provided for phone numbers, currency, percentage, precision, positional notation, and file size.
+See the [API
+Documentation](https://api.rubyonrails.org/classes/ActionView/Helpers/AssetTagHelper.html#method-i-stylesheet_link_tag)
+for more information.
 
-#### number_to_currency
+### video_tag
 
-Formats a number into a currency string (e.g., $13.65).
+Generate an HTML video tag with source(s), either as a single tag for a string
+source or nested source tags within an array for multiple sources. The `sources`
+can be full paths, files in your public videos directory, or Active Storage
+attachments.
 
 ```ruby
-number_to_currency(1234567890.50) # => $1,234,567,890.50
+video_tag("trailer")
+# => <video src="/videos/trailer"></video>
+
+video_tag(["trailer.ogg", "trailer.flv"])
+# => <video><source src="/videos/trailer.ogg" /><source src="/videos/trailer.flv" /></video>
+
+video_tag("trailer", controls: true)
+# => <video controls="controls" src="/videos/trailer"></video>
 ```
 
-See the [API Documentation](https://api.rubyonrails.org/classes/ActionView/Helpers/NumberHelper.html#method-i-number_to_currency) for more information.
+INFO: Internally, `video_tag` uses [`video_path` from the
+AssetUrlHelpers](https://api.rubyonrails.org/classes/ActionView/Helpers/AssetUrlHelper.html#method-i-video_path)
+to build the video path.
 
-#### number_to_human
+See the [API
+Documentation](https://api.rubyonrails.org/classes/ActionView/Helpers/AssetTagHelper.html#method-i-video_tag)
+for more information.
 
-Pretty prints (formats and approximates) a number so it is more readable by users; useful for numbers that can get very large.
+JavaScript
+----------------------------------------------
 
-```ruby
-number_to_human(1234)    # => 1.23 Thousand
-number_to_human(1234567) # => 1.23 Million
+A set of methods for working with JavaScript in your views.
+
+### escape_javascript
+
+Escapes carriage returns and single and double quotes for JavaScript segments.
+You would use this method to take a string of text and make sure that it doesn’t
+contain any invalid characters when the browser tries to parse it.
+
+For example, if you have a partial with a greeting that contains double quotes,
+you can escape the greeting to use in a JavaScript alert.
+
+**app/views/users/greeting.html.rb**
```

Comment:
More of a note for us to think about, I wanted to hear your thoughts: (maybe it's something we should also settle on a default)

In some cases like here in this guide, we're using a bold "paragraph" before the code block, which looks like this:

<details><summary>Details</summary>
<p>

<img width="459" alt="Screenshot 2024-04-08 at 17 12 48" src="https://github.com/rails/rails/assets/26328/3fddb978-089d-4426-a340-042a219e511c">

</p>
</details>

In some other cases, we have file paths that are inside the code block, as the first commented out line, or even multiple times to indicate where they are located: (this one is from the AR Basics guide)

<details><summary>Details</summary>
<p>

<img width="442" alt="Screenshot 2024-04-08 at 17 14 03" src="https://github.com/rails/rails/assets/26328/8e4b3a9c-b0c5-43b7-9384-d1c153d9375f">

</p>
</details> 

The former may distract the reader a bit, so I personally kind of like the commented version within the code block. Here's how this one in particular would look like:

<details><summary>Details</summary>
<p>

<img width="437" alt="Screenshot 2024-04-08 at 17 15 23" src="https://github.com/rails/rails/assets/26328/dca1beff-ac5e-4435-be21-f604d8c547b7">

</p>
</details> 

The drawback is that it gets copied over with the "copy" button, but so does the output of code we have in some other examples, etc., so I think that's fine.

Anyway, just wanted to hear your thoughts on this one (also cc'ing @bhumi1102 @p8)

Reply:
This is a good observation, and I think it's great that we're working towards consistency! My vote goes towards the commented version within the code block. I think it's less imposing and does not try to take the attention away from the code.

Reply:
> here's one bit of food for thought about the file paths in code samples that might be worth taking a look as well (maybe we can settle on a definitive way to do it going forward), but otherwise this will be ready to go.

For now, I'll change this to the commented version and we can update it across the guides once we've heard everyone's thoughts and decided on a way forward, if that's okay?

Reply:
> The drawback is that it gets copied over with the "copy" button, but so does the output of code we have in some other examples, etc., so I think that's fine.

The "copy" button script currently ignores some code, like the `$` and the output for commandline examples. 
An example is the ruby version code:
https://edgeguides.rubyonrails.org/getting_started.html#installing-ruby

https://github.com/rails/rails/blob/4c258caba245fbd311b0e8f157a9d8a5fc1c54b7/guides/rails_guides/markdown/renderer.rb#L85-L94

We could do the same for Ruby comments matching file paths.

Reply:
Yeah, I think that would be nice. Would you like to work on it?

Reply:
Sure. I can have a look.